CryptomatorFileProvider/NSFileProviderItemIdentifier+Database.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  NSFileProviderItemIdentifier+Database.swift
+//  CryptomatorFileProvider
+//
+//  Created by Philipp Schmid on 04.05.22.
+//  Copyright © 2022 Skymatic GmbH. All rights reserved.
+//
+
+import FileProvider
+import Foundation
+
+public extension NSFileProviderItemIdentifier {
+	static let rootContainerDatabaseValue: Int64 = 1
+	private static let delimiter: Character = ":"
+
+	/**
+	 Preferred constructor to create an `NSFileProviderItemIdentifier`.
+
+	 An `NSFileProviderItemIdentifier` has the format:
+	 `<domainIdentifier>:<itemID>`
+
+	 This ensures that the `NSFileProviderDomainIdentifier` can be derived from any `NSFileProviderItemIdentifier` (except the `.rootContainer` and `.workingSet`).
+	 This is necessary because the `extensionContext.domainIdentifier` in the FileProviderExtensionUI is not guaranteed to be the `domainIdentifier`
+	 of the currently visible `NSFileProviderDomain` and thus a correct XPC connection could not be established.
+	 */
+	init(domainIdentifier: NSFileProviderDomainIdentifier, itemID: Int64) {
```

Comment:
Food for thought: Would it be more consistent to call it `itemIdentifier` instead of `itemID`?

Reply:
Hm I'm not sure, although of course it has the same meaning, `itemIdentifier` is usually used to refer to an `NSFileProviderItemIdentifier` and `itemID` (or just `id`) is used to refer to the id of an `ItemMetadata` object.

Reply:
All right, works for me. 😁